### PR TITLE
graalvm and aws-lambda adds aws-custom-runtime

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ public void processSelectedFeatures(FeatureContext featureContext) {
             featureContext.addFeature(shadePlugin);
         }
 
-        if (featureContext.getBuildTool() == BuildTool.MAVEN && featureContext.isPresent(GraalVM.class)) {
+        if (featureContext.isPresent(GraalVM.class)) {
```
Comment (ilopmar) [inline starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java]: Why doing this? Adding `GraalVM` feature is only necessary when using Maven. With Gradle the plugin takes care of everything.

File: starter-core/src/test/groovy/io/micronaut/starter/feature/function/awslambda/AwsLambdaSpec.groovy [hunk 2]
```diff
@@ -227,7 +227,7 @@ class AwsLambdaSpec extends ApplicationContextSpec implements CommandOutputFixtu
     }
 
     @Unroll
-    void 'Application file is generated for a default application type with gradle and features aws-lambda and graalvm for language: #language'(Language language, String extension) {
+    void 'Application file is not generated for a default application type with gradle and features aws-lambda and graalvm for language: #language'(Language language, String extension) {
```
Comment (ilopmar) [inline starter-core/src/test/groovy/io/micronaut/starter/feature/function/awslambda/AwsLambdaSpec.groovy]: Why this change of behaviour?

**Review by ilopmar — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java (hunk 1 above)
Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java]: No, it does not. Use: 

Application Type:  Serverless function
Features: `aws-lambda`, `graalvm`

and: 

```
% ./gradlew buildNativeLambda

FAILURE: Build failed with an exception.

* What went wrong:
Task 'buildNativeLambda' not found in root project 'micronautguide'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 568ms
```

File: starter-core/src/test/groovy/io/micronaut/starter/feature/function/awslambda/AwsLambdaSpec.groovy (hunk 2 above)
Comment (sdelamo) [reply starter-core/src/test/groovy/io/micronaut/starter/feature/function/awslambda/AwsLambdaSpec.groovy]: See above

File: starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java (hunk 1 above)
Comment (ilopmar) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java]: Try with `./gradlew buildNativeLambda -Pmicronaut.runtime=lambda`

Comment (graemerocher) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java]: Yeah this shouldn't be necessary

Comment (graemerocher) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java]: It seems it should only be added if application type is server less function. Note generally

Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java]: This does not work either: 

```
./gradlew buildNativeLambda -Pmicronaut.runtime=lambda
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
Task 'buildNativeLambda' not found in root project 'micronautguide'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 5s
```

Comment (sdelamo) [reply starter-core/src/main/java/io/micronaut/starter/feature/function/awslambda/AwsLambda.java]: @graemerocher > It seems it should only be added if application type is server less function. Not generally.

I did that change.

**Pull request merged**